bundles/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/internal/DSCAlarmActiveBinding.java

```diff
@@ -859,21 +821,22 @@ private void setTimeStampState(String timeStamp) {
 
         itemName = getItemName(DSCAlarmItemType.PANEL_TIME_STAMP, 0, 0);
 
-        if (itemName != "") {
+        if (StringUtils.isNotEmpty(itemName)) {
             DSCAlarmBindingConfig config = getItemConfig(itemName);
 
             if (config != null) {
                 Item item = getItem(itemName);
                 if (item != null) {
-                    DSCAlarmDeviceProperties dsclarmDeviceProperties = dscAlarmItemUpdate.getDeviceProperties(item, config);
 
-                    if (dsclarmDeviceProperties != null) {
+                    State onOffState = item.getState();
 
-                        boolean isTimeStamp = dsclarmDeviceProperties.getSystemTimeStamp();
+                    if (onOffState instanceof OnOffType) {
+                        OnOffType value = (OnOffType) onOffState;
 
-                        if ((timeStamp == "" && isTimeStamp == false) || (timeStamp != "" && isTimeStamp == true)) {
+                        if ((timeStamp == "" && value.equals(OnOffType.OFF)) || (timeStamp != "" && value.equals(OnOffType.ON))) {
```

Comment:
Please change all instances of `== ""` and `!= ""` throughout the code to use a correct method of determining if the left-hand side of the expression is or is not the empty string.  One possibility is `StringUtils.isBlank` and `StringUtils.isNotBlank`.  If you statically import these methods, it can be reduced to `isBlank` and `isNotBlank`.  Thank you.